### ISSUE TITLE
[Feature Request] Sketcher: smart dimension constraint

### ISSUE THREAD

**Issue by 69Piero** [FreeCAD/FreeCAD#7218]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=8&t=70376

### Subproject(s) affected?

Sketcher

### Idea description

I feel good to enhance the dimensional constraints into a single smart tool that blends all the current ones and, by mouse direction, it becomes horizontal/vertical/distance(when the entity is a circle/arc obviously
it becomes radius/diameter.

Basically I mean a tool like other cads have(ie: Solidworks/Autocad/inventor), just one powerful command and one button that, moreover, cleans the screen from three or four buttons making the same intent. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by adrianinsaval** [comment]
This is a duplicate of #5582 please follow the guidelines and don't create feature requests without discussing first or without searching for existing issues or forum discussions (of which there are many).

#5582 didn't exactly follow the guidelines either unfortunately and the "discussion" is not linked and it's more a monologue than a discussion: https://forum.freecadweb.org/viewtopic.php?f=8&t=5711&p=46304&hilit=1416#p46304
